handbook/Best practices/TypeScript/Project-References.md

```diff
@@ -0,0 +1,166 @@
+# Project References
+
+## Table of contents
+
+1. [Why use Project References?](#why-use-project-referencesgi)
+1. [Migrating a codebase](#migrating-a-codebase)
+1. [Tips](#tips)
+1. [Caveats](#caveats)
+
+
+## Why use Project References?
+
+Project References provides built-in scalability for TypeScript. It does this by providing developers tools to separate their code into smaller blocks. This enforces logical guidelines across the codebase which also results in healthier code over time. 
+
+The logical guidelines enforced by Project References allow TypeScript tools to operate on smaller chunks of work at a time. This in turn improves responsiveness and tightens the feedback between the VSCode and the TypeScript Language Server. 
+
+
+## Migrating a codebase
+
+TypeScript's documentation and guidance on [Project References](https://www.typescriptlang.org/docs/handbook/project-references.html) are trivial to implement in new codebases. In larger codebases, leveraging Project References can be cumbersome. This section outlines our best practices on migrating a large codebase to gradually adopt Project References.
+
+
+### Understanding your codebase
+
+In most large codebases, migrating to Project References in one go is likely, not feasible. The migration would have to be done gradually over time. 
+
+We recommend starting out by laying a foundation for the rest of the codebase to consume. This foundation generally includes the most widely used parts of your codebase. A good place to start might be the `packages/`, `tests/` or `app/utilities`. 
+
+In our experience, we've noticed that some `app/utilties` may be tightly coupled with `tests/` utilties, `app` code and `packages/` . In some cases even resulting in circular dependencies. We suggest you identifying some of the these utilties and extracting them into isolated project references hosted in `packages/@<project>-utilties`.
+
+### Starting the foundation
+
+Start by creating an entrypoint for your project references. For this guide we'll assume our foundation that our app relies on consists of our projects `packages/` and `tests` directory. 
```

Comment:
> For this guide we'll assume our foundation that our app

this reads a bit clunky, maybe try to make it more concise like

> For simplicity, lets assume the foundation of our app consists of a `packages/` and `tests` directory.

TBH I'm not sure that "the foundation of our app" is a clear enough concept since it could be conflated with the `foundation` folder anyway though.

Reply:
How leaf nodes of our dependency tree?